core-bundle/src/Routing/ResponseContext/PartialResponseHeaderBag.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Routing\ResponseContext;
+
+use Symfony\Component\HttpFoundation\ResponseHeaderBag;
+
+/**
+ * Symfony's core ResponseHeaderBag forces a Date header to be present
+ * which in case of the ResponseContext handling is not required.
+ * We have to ensure a completely empty header bag.
+ */
+class PartialResponseHeaderBag extends ResponseHeaderBag
+{
+    public function __construct(array $headers = [])
+    {
+        parent::__construct($headers);
+
+        $this->remove('cache-control');
+        $this->remove('date');
+    }
+
+    /**
+     * @param string $key
+     */
+    public function remove($key): void
+    {
+        parent::remove($key);
+
+        $uniqueKey = strtr($key, self::UPPER, self::LOWER);
```

Comment:
This will trigger a warning if `$key` is not a string in PHP 8.

Reply:
It's copied from Symfony Core. `$key` must be a string according to the docs ;)

Reply:
It is still not enforced, therefore we either have to handle the `!is_string($key)` case or cast the variable explicitly.

Reply:
It's not enforced in Symfony either. The phpdoc clearly state `@param string`. If you pass something else then it doesn't work - yes :D In newer versions Symfony has a type hint anway but I couldn't use that because of being compatible with lower versions. But feel free to add a `(string)` type cast if you prefer that.

Reply:
@ausi WDYT? Should we be defensive here or is it OK to handle it like Symfony does?

Reply:
I would keep it as it is because with Symfony 5 the `parent::remove($key)` would already trigger the type error and we would be best compatible with Symfony this way IMO.